core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -228,6 +228,12 @@ private TableProperties() {}
   public static final String ORC_BATCH_SIZE = "read.orc.vectorization.batch-size";
   public static final int ORC_BATCH_SIZE_DEFAULT = 5000;
 
+  public static final String PLANNING_MODE = "read.planning-mode";
+  public static final String PLANNING_MODE_DEFAULT = PlanningMode.AUTO.modeName();
+
+  public static final String DELETE_PLANNING_MODE = "read.delete.planning-mode";
+  public static final String DELETE_PLANNING_MODE_DEFAULT = PlanningMode.LOCAL.modeName();
```

Comment:
I am debating whether these configs make sense at the table level. Using `LOCAL` for deletes by default as those are less likely to be filtered using min/max filters and we will need to load stats so the cost of bringing them to the driver will be higher.

Reply:
I think that it does make sense to have these as table config because this is dependent on the table structure. If a table has a ton of deletes, you'd probably want to set this to AUTO to override. I'm not sure whether we'd want this to also be a default in Spark though. Seems like we may want to override, but not actually get the default value from here.